src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetManagedTemplatePackage.cs

```diff
@@ -156,6 +112,24 @@ public DateTime LastChangeTime
             }
         }
 
+        public string? Trusted
+        {
+            get => Details.TryGetValue(TrustedKey, out string trusted) ? trusted : false.ToString();
+            set => Details.AddOrRemove(TrustedKey, value!, InsertionCondition);
```

Comment:
the intention of this method is not clear. please try to rephrase.
also this won't work in case value is null - the exception will be thrown @ `value!`.